src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +80,14 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+  
+  /** Returns the total number of messages for all users.
+   * 
+   *  @return total number of messages posted by all users, limited to 1000.
+   *  */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
I read some more about this and it looks like the reasoning behind the 1000 limit is indeed that the query time grows linearly with the number of entities. 

If you want to count the number of entities in real time, then it looks like the recommended thing is to use pagination (let me know if I should explain that more).

If you don't actually care about getting _exactly_ the number of messages that exist, but a fairly close estimate, there's apparently a `Datastore Statistics` thing you can use. Here's a link to a StackOverflow answer explaining how to use the Datastore Statistics if you're interested: https://stackoverflow.com/a/10471802/2681057

Since it will be a while until we have 1,000 messages, I think this is fine as it is unless you're looking to take on something adventurous.